Library/Homebrew/cmd/--env.rb

```diff
@@ -3,19 +3,37 @@
 
 require "extend/ENV"
 require "build_environment"
+require "utils/shell"
 
 module Homebrew
+  def explicit_shell
+    shell = ARGV.value("shell")
+    return :noshell if shell == "none"
```

Comment:
`explicit_shell` returns `nil` when no shell has been passed in through the command line and `:noshell` when `--shell=none` has been specified. `--shell=none` is probably not the right way to do this but I wanted some way to force the `key: value` format. Originally I was using `--shell=`, but `ARGV.value` returns nil when an explicit empty value is passed and when the option is not passed at all.